packages/@yoda/network/index.js

```diff
@@ -0,0 +1,141 @@
+
+class NetworkAgent extends EventEmitter {
+  constructor () {
+    super()
+
+    this._remoteCallTarget = 'net_manager'
+    this._remoteCallCommand = 'network.command'
+    this._remoteCallTimeout = 60 * 1000
+
+    this._flora = new FloraComp('@network')
+    this._flora.init()
+  }
+
+  subscribe () {
+    this._flora.subscribe('network.status', (caps, type) => {
+      var msg = JSON.parse(caps[0])
+
+      if (msg.network) {
+        this.emit('network.status', msg.network)
+      } else if (msg.wifi) {
+        this.emit('wifi.status', msg.wifi)
+      } else if (msg.ethernet) {
+        this.emit('ethernet.status', msg.ethernet)
+      } else if (msg.modem) {
+        this.emit('modem.status', msg.modem)
+      }
+    })
+  }
+
+  _remoteCall (device, command, params) {
+    var data = {
+      device: device,
+      command: command
+    }
+    if (params) { data.params = params }
+
+    return this._flora.call(
+      this._remoteCallCommand,
+      [JSON.stringify(data)],
+      this._remoteCallTarget,
+      this._remoteCallTimeout
+    ).then(reply => {
+      return Promise.resolve(JSON.parse(reply.msg[0]))
+    }, err => {
+      return Promise.reject(err)
+    })
+  }
+
+  triggerStatus () {
+    return this._remoteCall('NETWORK', 'TRIGGER_STATUS')
+  }
+
+  capacities () {
+    return this._remoteCall('NETWORK', 'GET_CAPACITY')
+  }
+
+  getStatus (device) {
+    return this._remoteCall(device, 'GET_STATUS')
+  }
+
+  getNetworkStatus () {
+    return this.getStatus('NETWORK').then(reply => {
+      return Promise.resolve(reply.network)
+    }, err => {
+      return Promise.reject(err)
+    })
+  }
+
+  getWifiStatus () {
+    return this.getStatus('WIFI').then(reply => {
+      return Promise.resolve(reply.wifi)
+    }, err => {
+      return Promise.reject(err)
+    })
+  }
+
+  wifiOpen (ssid, passwd) {
+    return this._remoteCall('WIFI', 'CONNECT', {'SSID': ssid, 'PASSWD': passwd})
+  }
+
+  wifiClose () {
+    return this._remoteCall('WIFI', 'DISCONNECT')
+  }
+
+  wifiStartScan () {
+    return this._remoteCall('WIFI', 'START_SCAN')
+  }
+
+  wifiStopScan () {
+    return this._remoteCall('WIFI', 'STOP_SCAN')
+  }
+
+  wifiScanList () {
+    return this._remoteCall('WIFI', 'GET_WIFILIST')
+  }
+
+  wifiRemove (ssid, passwd) {
+    return this._remoteCall('WIFI', 'REMOVE', {'SSID': ssid, 'PASSWD': passwd})
+  }
+
+  wifiConfigNum () {
+      return this._remoteCall('WIFI', 'GET_CFG_NUM')
+  }
+
+  wifiApOpen (ssid, passwd, ip, timeout) {
+    return this._remoteCall('WIFI_AP', 'CONNECT', {
+      SSID: ssid,
+      PASSWD: passwd,
+      IP: ip,
+      TIMEOUT: timeout
+    })
+  }
+
+  wifiApClose () {
+    return this._remoteCall('WIFI_AP', 'DISCONNECT')
+  }
+
+  modemOpen () {
+    return this._remoteCall('MODEM', 'CONNECT')
+  }
+
+  modemClose () {
+    return this._remoteCall('MODEM', 'DISCONNECT')
+  }
+}
+
+module.exports = {
+  CONNECTED: 'CONNECTED',
+  DISCONNECTED: 'DISCONNECTED',
```

Comment:
How about exposing a method `isConnected()` on `NetworkAgent` instead of the above?